java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -568,6 +568,16 @@ private void indexLoadedRepo(final RepositoryInfo repo, boolean updateLocal) thr
                         // IllegalArgumentException signals remote archive format problems
                         fetchFailed = true;
                         throw new IOException("Failed to load maven-index for: " + indexingContext.getRepositoryUrl(), ex);
+                    } catch (RuntimeException ex) {
+                        // thread pools, like the one used in maven-indexer's IndexDataReader, may suppress cancellation exceptions
+                        // lets try to find them again
+                        if (isCancellation(ex)) {
+                            Cancellation cancellation = new Cancellation();
+                            cancellation.addSuppressed(ex);
+                            throw cancellation;
+                        } else {
+                            throw ex;
+                        }
```

Comment:
we can do this now since it was fixed upstream in 7.0.2: https://github.com/apache/maven-indexer/pull/314